spec/jmap/push.mdown

```diff
@@ -26,25 +31,39 @@ Refer to the Authentication section of this spec for details on how to get the U
 
-The JMAP server MUST follow any redirects. If the final response code from the server is `2xx`, the callback is considered a success. If the response code is `503` (Service Unavailable), the JMAP server MAY try again later (but may also just drop the event). If the response code is `429` (Too Many Requests) the JMAP server SHOULD attempt to reduce the frequency of pushes to that URL. Any other response code MUST be considered a **permanent failure** and the callback should be deregistered (not tried again even for future events unless explicitly re-registered by the client).
+The request MUST have a content type of `application/json` and contain the same UTF-8 JSON encoded object as described above as the body.
 
-The URL set by the client MUST use the HTTPS protocol and SHOULD encode within it a unique token that can be verified by the server to know that the request comes from the JMAP server the authenticated client connected to.
+The JMAP server MUST set the following header in the POST request:
+
+- `TTL`: A value in seconds that suggests how long a push message is retained by the push service, as per [@!RFC8030] section 5.2.
+
+The JMAP server MAY include the following headers in the POST request:
+
+- `Urgency`: One of `very-low`, `low`, `normal`, or `high` as per [@!RFC8030]
+  section 5.3.
+- `Topic`: A token that encodes the set of types included in the *changed*
+  property of this push request. This MUST be restricted to no more than 32
+  characters from the URL and filename-safe Base 64 alphabet ([@!RFC4648]), as
+  per [@!RFC8030] section 5.4. This allows the push server to replace
+  undelivered messages which are entirely superceded by the later message.
+
+The JMAP server MUST follow any redirects. If the final response code from the server is `2xx`, the callback is considered a success. In the case of `4xx` or `5xx` responses, the server SHOULD try to do
+
+If the response code is `503` (Service Unavailable), the JMAP server MAY try again later (but may also just drop the event). If the response code is `429` (Too Many Requests) the JMAP server SHOULD attempt to reduce the frequency of pushes to that URL. Any other `4xx` or `5xx` response code MUST be considered a **permanent failure** and the callback should be deregistered (not tried again even for future events unless explicitly re-registered by the client).
 
 The callback is tied to the access token used to create it. Should the access token expire or be revoked, the callback MUST be removed by the JMAP server. The client MUST re-register the callback after reauthenticating to resume callbacks.
 
 Each session may only have a single callback URL registered. It can be set or retrieved using the following API calls.
 
 ### setPushCallback
 
-To set the web hook, make a call to *setPushCallback*. It takes the following argument:
+To set the callback URL, make a call to *setPushCallback*. It takes the following argument:
 
 - **callback**: `String|null`
```

Comment:
This isn't sufficient.  See https://w3c.github.io/push-api/#dom-pushsubscription for a description of the properties that are needed to use (most implementations of) RFC 8030.